chapters/0-introduction.tex

```diff
@@ -0,0 +1,266 @@
+Code today is no longer a means to an end (such as a research paper),
+rather it is part of the output itself: a means for communicating how something was done,
+in a world where the credibility and transparency of data collection, cleaning, and analysis is increasingly important.
+As this is fundamental to the remainder of the book's content,
+we provide here a brief introduction using code for \textbf{process standardization} in data collection.
+
+Process standardization means that there is
+little ambiguity about how something ought to be done,
+and therefore the tools to do it can be set in advance.
+Standard processes for code help other people to read your code.\sidenote{
+\url{https://dimewiki.worldbank.org/Stata_Coding_Practices}}
+Code should be well-documented, contain extensive comments, and be readable in the sense that others can:
+(1) quickly understand what a portion of code is supposed to be doing;
+(2) evaluate whether or not it does that thing correctly; and
+(3) modify it efficiently either to test alternative hypotheses
+or to adapt into their own work.\sidenote{\url{https://kbroman.org/Tools4RR/assets/lectures/07_clearcode.pdf}}
+
+In the context of data collection, there are many places where good code and code annotation
+are important to accurately convey and implement the intended research design in reality.
+For example, statistical software is now an essential part of
+design components such as sampling, randomization, and power analysis.
+Being able to accurately and reproducibily implement these tasks
+is essential to the success and credibility of any modern randomized experiment.
+Simultaneously, researchers need to maintain records of the structure of their data,
+which involves managing and collating different types of information,
+often at different levels of analysis and different stages in time.
+Keeping a clear, human-readable record of these code and data structures is critical.
+Finally, as data is collected, code tools must be used to validate its quality,
+organize and manage the linkages between datasets,
+resolve and correct errors that occur in the process of data acquisition,
+and, ultimately, create and analyze the measures that
+are the motivation for the research study.
+
+A breakdown in any part of this data pipeline
+means that the data that are acquired become unreliable.
+If that happens, the results cannot be faithfully interpreted
+as being an accurate picture of the intended research design.
+Because we almost never have ``laboratory'' settings in this type of research,
+such a failure has a very high cost:
+we will have wasted the investments that were made into knowledge generation,
+with little ability to reproduce or recreate the situation
+where we intended to operate the research project.
+Hence accurate and reproducible data management is a core component
+of the credibility of development research.
+
+\section{Outline of this book}
```

Comment:
it feels like this comes late in the intro. or that at least we might want to clarify very early how the overall structure follows the project cycle, say up in the section where we explain what the book aims to do.